### PR TITLE
[kube-prometheus-stack] Update prometheus-operator to v0.44.0

### PR DESCRIPTION
**Pull request by sev3ryn**
#### What this PR does / why we need it:
update to prometheus operator version v0.44.0

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by bismarck — CHANGES_REQUESTED** [review]
@sev3ryn Thanks for the PR. I was actually testing this version of prometheus-operator locally.  The `--logtostderr` [flag](https://github.com/sev3ryn/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L61) got [removed](https://github.com/prometheus-operator/prometheus-operator/pull/3630) from the operator (and the pods will fail to start if it is present), could you add a conditional for this flag for < v0.44.0?

File: charts/kube-prometheus-stack/scripts/update_crds.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -eu
```
Comment (bismarck) [inline charts/kube-prometheus-stack/scripts/update_crds.sh:1]: Mind moving this script to the hack directory?

**Review by bismarck — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/scripts/update_crds.sh (hunk 1 above)
Comment (sev3ryn) [reply charts/kube-prometheus-stack/scripts/update_crds.sh:1]: sure, will do

**Comment by sev3ryn** [comment]
@bismarck addressed your comments in 6993936

**Review by bismarck — APPROVED** [review]
lgtm

**Pull request merged**